adx/resource_adx_table_update_policy.go

```diff
@@ -87,7 +94,13 @@ func resourceADXTableUpdatePolicyCreateUpdate(ctx context.Context, d *schema.Res
 	transactional := d.Get("transactional").(bool)
 	propagateIngestionProperties := d.Get("propagate_ingestion_properties").(bool)
 
-	createStatement := fmt.Sprintf(".alter table %s policy update @'[{\"IsEnabled\": %t, \"Source\": \"%s\", \"Query\": \"%s\", \"IsTransactional\": %t, \"PropagateIngestionProperties\": %t}]'", tableName, enabled, sourceTable, query, transactional, propagateIngestionProperties)
+	var createStatement string
+	if len(d.Get("managed_identity").(string)) > 0 {
+		managedIdentity := d.Get("managed_identity").(string)
+		createStatement = fmt.Sprintf(".alter table %s policy update @'[{\"IsEnabled\": %t, \"Source\": \"%s\", \"Query\": \"%s\", \"IsTransactional\": %t, \"PropagateIngestionProperties\": %t, \"ManagedIdentity\": \"%s\"}]'", tableName, enabled, sourceTable, query, transactional, propagateIngestionProperties, managedIdentity)
+	} else {
+		createStatement = fmt.Sprintf(".alter table %s policy update @'[{\"IsEnabled\": %t, \"Source\": \"%s\", \"Query\": \"%s\", \"IsTransactional\": %t, \"PropagateIngestionProperties\": %t}]'", tableName, enabled, sourceTable, query, transactional, propagateIngestionProperties)
+	}
```

Comment:
```suggestion
	createStatement = fmt.Sprintf(".alter table %s policy update @'[{\"IsEnabled\": %t, \"Source\": \"%s\", \"Query\": \"%s\", \"IsTransactional\": %t, \"PropagateIngestionProperties\": %t}]'", tableName, enabled, sourceTable, query, transactional, propagateIngestionProperties)
	
	if len(d.Get("managed_identity").(string)) > 0 {
		managedIdentity := d.Get("managed_identity").(string)
		createStatement = fmt.Sprintf(".alter table %s policy update @'[{\"IsEnabled\": %t, \"Source\": \"%s\", \"Query\": \"%s\", \"IsTransactional\": %t, \"PropagateIngestionProperties\": %t, \"ManagedIdentity\": \"%s\"}]'", tableName, enabled, sourceTable, query, transactional, propagateIngestionProperties, managedIdentity)
	}
```

Minor nitpick remark, just override the default value if `managed_identity` is set, instead of if/else?